torch_xla/core/xla_model.py

```diff
@@ -376,14 +457,73 @@ def _fetch_gradients(optimizer):
 
 
 def _get_all_reduce_token():
-  token = getattr(_TLS, 'all_reduce_token', None)
+  devctx = _get_device_context()
+  token = getattr(devctx, 'all_reduce_token', None)
   if token is None:
     token = torch_xla._XLAC._xla_create_token()
-    _TLS.all_reduce_token = token
-  return token
+    devctx.all_reduce_token = token
+  return token, devctx
+
+
+def _torch_all_reduce(reduce_type, inputs, group=None):
```

Comment:
You can remove from here ... (sea-of-devices)

Reply:
Done.